jme3-lwjgl/src/main/java/com/jme3/input/lwjgl/JInputJoyInput.java

```diff
@@ -262,7 +263,7 @@ protected void addButton( Component comp ) {
                 original = String.valueOf(buttonIndex.size());
             }
             String logicalId = JoystickCompatibilityMappings.remapButton( controller.getName(), original );
-            if( logicalId != original ) {
+            if (logger.isLoggable(Level.FINE) && !Objects.equals(logicalId, original)) {
```

Comment:
This may fix a logic error in the existing code.

Reply:
Regarding the original code, it is difficult to determine it if works or not - which is already a problem.
So I would recommend to replace it with code that is obviously correct.
Since this change is related only to the display of the log message, I chose to include it in this PR.

Please tell me explicitely if you want me to preserve the original logic.

Reply:
Normally I fuss about changes outside the scope of the PR, but in this case, I don't mind. I commented in order to bring it to the attention of other reviewers.

I don't want you to preserve the original logic.

Reply:
Note: the change is fine but the reason it "worked" before is because the strings are regurgitated by the library and so should always be the same instance from call to call.  Meaning: code never constructs one of these strings.  We are passed them and then in turn pass them around.

...it still looks weird and I'm ok with the change.  Though consider that it is now not catching the same condition that it did before.  Whether that matters or not, I don't know.

My explanation above on how the logical IDs are handled could in theory mean that other code is counting on this instance-sameness and this check was confirming that (and now doesn't anymore).  If that's the case, I would certainly have expected a comment.